src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -135,14 +135,19 @@ internal void ReleaseConnection()
                 {
                     if (!_isClosed && _tcpClient != null)
                     {
-                        //free cbt buffer
-                        if (_channelBindingToken != null)
+                        try
                         {
-                            _channelBindingToken.Close();
+                            // Gracefully close the transmission channel
+                            _tcpClient.Client.Blocking = false;
+                            QuitCommand.Send(this);
```

Comment:
This code can now throw. We should catch some exceptions here.
The code in Abort is missing some catch clauses too (see https://github.com/dotnet/corefx/issues/40711).
Since the code in Abort is very similar to this, maybe the Abort method can call into this one? This method can deal with the differences using a `bool abort = false` argument.

Reply:
> This code can now throw. We should catch some exceptions here.

Why?Abort can be called "in the middle" and the semantic of "aborts" suggest a good chance of exception due to middle states.
Should we hide also exception here?You're suggesting to avoid to bubble up exception for quit? cc: @scalablecory 

> The code in Abort is missing some catch clauses too (see dotnet/corefx#40711).
Since the code in Abort is very similar to this, maybe the Abort method can call into this one? This method can deal with the differences using a bool abort = false argument.

I see, yep we could merge code.

Reply:
> Should we hide also exception here?

Eating exceptions is reasonable here as it would not result in a corrupt state for the user -- either way the socket is getting closed and the server is expected to behave properly. Additionally, the user is probably not expecting an exception here.

> Since the code in Abort is very similar to this, maybe the Abort method can call into this one? This method can deal with the differences using a `bool abort = false` argument.

Merging the two makes sense. I've no preference for this being done as part of this PR or as a different issue.

Reply:
Added `IOException` catch for network error and `InvalidOperationException`(and subclassed `ObjectDisposedException`) added for `QuitCommand.Send`